### PR TITLE
imxrt: use UART_CONSOLE to setup early console

### PR DESCRIPTION
**Pull request by gerard5**
JIRA: RTOS-118

## Description

Handle BOARD_CONFIG variables like `UART_CONSOLE=n`, `UARTn_BAUDRATE`
to setup early console based on common to `imxrt-multi` and `plo`
setup.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt1176, imxrt1064).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/imxrt/117x/console.c [hunk 1]
```diff
@@ -19,8 +19,17 @@
...
 #include "../../include/arch/imxrt1170.h"
 
-#ifndef CONSOLE
-#define CONSOLE 11
+#define CONCAT(a, b)         a##b
+#define CONCAT2(a, b)        CONCAT(a, b)
+#define UART_PIN(n, pin)     CONCAT(UART, n##_##pin)
+#define CONSOLE_MUX(n, pin)  (CONCAT2(pctl_mux_gpio_, UART_PIN(n, pin)))
+#define CONSOLE_PAD(n, pin)  (CONCAT2(pctl_pad_gpio_, UART_PIN(n, pin)))
+#define CONSOLE_BAUD(n)      (UART_PIN(n, BAUDRATE))
```
[the remark this answers is not in the document]
Comment (gerard5) [reply hal/armv7m/imxrt/117x/console.c]: Fixed.

File: hal/armv7m/imxrt/117x/console.c [hunk 2]
```diff
@@ -54,50 +63,72 @@ void hal_consolePrint(int attr, const char *s)
...
-		{ ((void *)0x40088000), pctl_clk_lpuart4 },
-		{ ((void *)0x4008c000), pctl_clk_lpuart5 },
-		{ ((void *)0x40090000), pctl_clk_lpuart6 },
-		{ ((void *)0x40094000), pctl_clk_lpuart7 },
-		{ ((void *)0x40098000), pctl_clk_lpuart8 },
-		{ ((void *)0x4009c000), pctl_clk_lpuart9 },
-		{ ((void *)0x400a0000), pctl_clk_lpuart10 },
-		{ ((void *)0x40c24000), pctl_clk_lpuart11 },
-		{ ((void *)0x40c28000), pctl_clk_lpuart12 }
+		{ ((void *)0x4007c000), pctl_clk_lpuart1, pctl_mux_gpio_ad_24, pctl_pad_gpio_ad_24, pctl_mux_gpio_ad_25, pctl_pad_gpio_ad_25, pctl_isel_lpuart1_txd, 0,  pctl_isel_lpuart1_rxd, 0 },
```
[the remark this answers is not in the document]
Comment (gerard5) [reply hal/armv7m/imxrt/117x/console.c]: Fixed.

**Review by MaciejPurski — DISMISSED** [review]
LGTM. hw tested on imxrt106x.

File: hal/armv7m/imxrt/117x/console.c [hunk 3]
```diff
@@ -13,14 +13,27 @@
...
 #include "../../include/errno.h"
 #include "../../include/arch/imxrt1170.h"
 
-#ifndef CONSOLE
-#define CONSOLE 11
+#define CONCAT(a, b)        a##b
+#define CONCAT2(a, b)       CONCAT(a, b)
+#define UART_PIN(n, pin)    CONCAT(UART, n##_##pin)
+#define CONSOLE_MUX(n, pin) (CONCAT2(pctl_mux_gpio_, UART_PIN(n, pin)))
+#define CONSOLE_PAD(n, pin) (CONCAT2(pctl_pad_gpio_, UART_PIN(n, pin)))
```
Comment (agkaminski) [inline hal/armv7m/imxrt/117x/console.c:30]: Either use these in `info` or delete these

Comment (gerard5) [reply hal/armv7m/imxrt/117x/console.c]: I've removed unused macros.

File: hal/armv7m/imxrt/10xx/console.c [hunk 4]
```diff
@@ -17,17 +17,21 @@
...
 #include "cpu.h"
 #include "console.h"
 #include "imxrt10xx.h"
 #include "../../../../include/errno.h"
 #include "../../../../include/arch/imxrt.h"
 
-#define UART_CLK 80000000
+#define CONCAT(a, b)        a##b
+#define UART_PIN(n, pin)    CONCAT(UART, n##_##pin)
+#define CONSOLE_BAUD(n)     (UART_PIN(n, BAUDRATE))
```
[the remark this answers is not in the document]
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c]: Done.

**Review by HBuczynski — DISMISSED** [review]

File: hal/armv7m/imxrt/10xx/console.c [hunk 5]
```diff
@@ -17,17 +17,21 @@
...
 #include "cpu.h"
 #include "console.h"
 #include "imxrt10xx.h"
 #include "../../../../include/errno.h"
 #include "../../../../include/arch/imxrt.h"
 
-#define UART_CLK 80000000
+#define CONCAT(a, b)     a##b
+#define UART_PIN(n, pin) CONCAT(UART, n##_##pin)
+#define CONSOLE_BAUD(n)  (UART_PIN(n, BAUDRATE))
```
Comment (agkaminski) [inline hal/armv7m/imxrt/10xx/console.c:28]: Perhaps:

```suggestion
#define CONSOLE_BAUD(n)  (UART##n##_BAUDRATE))
```

Comment (agkaminski) [reply hal/armv7m/imxrt/10xx/console.c:28]: So I think we should be at least able ro reduce this to 2 macros instead of 3, shouldn't we?

Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c]: Yeah, that's gonna work. Pushed an updated commit.

**Review by agkaminski — APPROVED** [review]

**Pull request merged**